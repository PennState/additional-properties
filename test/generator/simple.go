package acceptance

// Simple is the most basic struct to which we can add additional-
// properties.
type Simple struct {
	FieldA string                 `json:"fieldA"`
	AP     map[string]interface{} `json:"*"`
}

func newZeroSimple() interface{} {
	return &Simple{}
}

func newTestSimple() interface{} {
	return &Simple{
		FieldA: "Field A",
		AP: map[string]interface{}{
			"fieldB": "Field B",
			"fieldC": "Field C",
		},
	}
}

func newTestSimpleWithoutAP() interface{} {
	return &Simple{
		FieldA: "Field A",
	}
}
