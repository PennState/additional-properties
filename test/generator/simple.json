{"a":"Field A","b":"Field B","c":"Field C"}