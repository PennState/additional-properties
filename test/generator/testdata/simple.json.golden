{"fieldA":"Field A","fieldB":"Field B","fieldC":"Field C"}